src/main/java/dev/j3fftw/litexpansion/machine/WaterSprinkler.java

```diff
@@ -0,0 +1,152 @@
+
+import org.bukkit.Material;
+import org.bukkit.Particle;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.block.data.Ageable;
+import org.bukkit.inventory.ItemStack;
+
+import dev.j3fftw.litexpansion.Items;
+import dev.j3fftw.litexpansion.utils.Constants;
+import io.github.thebusybiscuit.slimefun4.api.MinecraftVersion;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunItems;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import io.github.thebusybiscuit.slimefun4.implementation.items.electric.machines.AbstractGrowthAccelerator;
+import io.github.thebusybiscuit.slimefun4.implementation.items.electric.machines.CropGrowthAccelerator;
+import io.github.thebusybiscuit.slimefun4.utils.ChestMenuUtils;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.api.BlockStorage;
+import me.mrCookieSlime.Slimefun.api.inventory.BlockMenu;
+import me.mrCookieSlime.Slimefun.cscorelib2.item.CustomItem;
+
+/**
+ * The {@link WaterSprinkler} speeds up the growth of nearby crops
+ * when water is under the machine
+ * Essentially a modified {@link CropGrowthAccelerator}
+ *
+ * @author FluffyBear
+ */
+public class WaterSprinkler extends AbstractGrowthAccelerator {
+
+    private final Set<Material> crops = EnumSet.noneOf(Material.class);
+    public static final int ENERGY_CONSUMPTION = 16;
+    public static final int CAPACITY = 64;
+    private static final int RADIUS = 2;
+    private static final int PROGRESS_SLOT = 4;
+    private static final CustomItem noWaterItem = new CustomItem(Material.BUCKET,
+        "&cNo water found",
+        "",
+        "&cPlease place water under the sprinkler!"
+    );
+    private static final CustomItem waterFoundItem = new CustomItem(Material.WATER_BUCKET,
+        "&bWater detected"
+    );
+
+    public WaterSprinkler() {
+        super(Items.LITEXPANSION, Items.WATER_SPRINKER, RecipeType.ENHANCED_CRAFTING_TABLE,
+            new ItemStack[] {
+                Items.REFINED_IRON, SlimefunItems.ELECTRIC_MOTOR, Items.REFINED_IRON,
+                new ItemStack(Material.BUCKET), Items.MACHINE_BLOCK, new ItemStack(Material.BUCKET),
+                Items.REFINED_IRON, SlimefunItems.MEDIUM_CAPACITOR, Items.REFINED_IRON
+            });
+
+        createPreset(this, Items.WATER_SPRINKER.getImmutableMeta().getDisplayName().orElse("&bWater Sprinkler"),
+            blockMenuPreset -> {
+                for (int i = 0; i < 9; i++)
+                    blockMenuPreset.addItem(i, ChestMenuUtils.getBackground(), ChestMenuUtils.getEmptyClickHandler());
+
+                blockMenuPreset.addItem(PROGRESS_SLOT, noWaterItem);
+            });
+
+        crops.add(Material.WHEAT);
+        crops.add(Material.POTATOES);
+        crops.add(Material.CARROTS);
+        crops.add(Material.NETHER_WART);
+        crops.add(Material.BEETROOTS);
+        crops.add(Material.COCOA);
+        crops.add(Material.SUGAR_CANE);
```

Comment:
what about melons stems and pumpkins stems not the growth of pumpkins and melons it self just the stem